packages/protocol/src/action/request/MalformedRequestError.ts

```diff
@@ -0,0 +1,12 @@
+/**
+ * @license
+ * Copyright 2022-2025 Matter.js Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { MatterError } from "#general";
+
+/**
+ * Thrown when an action request does not adhere to the Matter specification.
+ */
+export class MalformedRequestError extends MatterError {}
```

Comment:
Directly map to StatusResponseError for simplicity or do you plan error type conversions?

Reply:
This is a client-side error, it's thrown by the builder API if there's something illegal we can't prevent with types.  I updated the comment and made it extend `ImplementationError` instead